cli/slsa-verifier/main_regression_test.go

```diff
@@ -1795,6 +1795,99 @@ func Test_runVerifyNpmPackage(t *testing.T) {
 	}
 }
 
+func Test_runVerifyVSA(t *testing.T) {
+	// We cannot use t.Setenv due to parallelized tests.
+	os.Setenv("SLSA_VERIFIER_EXPERIMENTAL", "1")
+	t.Parallel()
+
+	tests := []struct {
+		name              string
+		AttestationPath   *string
+		subjectDigests    *[]string
+		verifierID        *string
+		resourceURI       *string
+		verifiedLevels    *[]string
+		publicKeyPath     *string
+		publicKeyID       *string
+		publicKeyHashAlgo *string
+		err               error
+	}{
+		{
+			name:              "success: gke",
+			AttestationPath:   PointerTo("gce/v1/gke-gce-pre.bcid-vsa.jsonl"),
+			subjectDigests:    PointerTo([]string{"gce_image_id:8970095005306000053"}),
+			verifierID:        PointerTo("https://bcid.corp.google.com/verifier/bcid_package_enforcer/v0.1"),
+			resourceURI:       PointerTo("gce_image://gke-node-images:gke-12615-gke1418000-cos-101-17162-463-29-c-cgpv1-pre"),
+			verifiedLevels:    PointerTo([]string{"BCID_L1", "SLSA_BUILD_LEVEL_2"}),
+			publicKeyPath:     PointerTo("gce/v1/vsa_signing_public_key.pem"),
+			publicKeyID:       PointerTo("keystore://76574:prod:vsa_signing_public_key"),
+			publicKeyHashAlgo: PointerTo("SHA256"),
+		},
+		{
+			name:              "success: gke, default public key hash algo",
+			AttestationPath:   PointerTo("gce/v1/gke-gce-pre.bcid-vsa.jsonl"),
+			subjectDigests:    PointerTo([]string{"gce_image_id:8970095005306000053"}),
+			verifierID:        PointerTo("https://bcid.corp.google.com/verifier/bcid_package_enforcer/v0.1"),
+			resourceURI:       PointerTo("gce_image://gke-node-images:gke-12615-gke1418000-cos-101-17162-463-29-c-cgpv1-pre"),
+			verifiedLevels:    PointerTo([]string{"BCID_L1", "SLSA_BUILD_LEVEL_2"}),
+			publicKeyPath:     PointerTo("gce/v1/vsa_signing_public_key.pem"),
+			publicKeyID:       PointerTo("keystore://76574:prod:vsa_signing_public_key"),
+			publicKeyHashAlgo: PointerTo(""),
+		},
+		{
+			name:              "fail: gke, unsupported public key hash algo",
+			AttestationPath:   PointerTo("gce/v1/gke-gce-pre.bcid-vsa.jsonl"),
+			publicKeyPath:     PointerTo("gce/v1/vsa_signing_public_key.pem"),
+			publicKeyHashAlgo: PointerTo("SHA123"),
+			err:               serrors.ErrorInvalidHashAlgo,
+		},
+		{
+			name:              "fail: gke, wrong public key hash algo",
+			AttestationPath:   PointerTo("gce/v1/gke-gce-pre.bcid-vsa.jsonl"),
+			publicKeyPath:     PointerTo("gce/v1/vsa_signing_public_key.pem"),
+			publicKeyID:       PointerTo(""),
+			publicKeyHashAlgo: PointerTo("SHA512"),
+			err:               serrors.ErrorNoValidSignature,
+		},
+		{
+			name:              "fail: gke, wrong key id",
+			AttestationPath:   PointerTo("gce/v1/gke-gce-pre.bcid-vsa.jsonl"),
+			publicKeyPath:     PointerTo("gce/v1/vsa_signing_public_key.pem"),
+			publicKeyID:       PointerTo("my_key_id"),
+			publicKeyHashAlgo: PointerTo("SHA256"),
+			err:               serrors.ErrorNoValidSignature,
+		},
+		// TODO: Add more tests for different scenarios.
```

Comment:
Do you have other tests you want to add? Otherwise I'd remove.

Reply:
Added some more tests